Problem2.java

```diff
@@ -1 +1,137 @@
+/*
+    1. Min HEap also known as priority queue.
+    2. Min heap is a complete binary tree.
+    3. Root node of a min heap must be the minimum value. 
+    4. The Heap data structure can be used to efficiently find the kth smallest element
+*/
+//Space Complexity O(n)
+class Minheap
+{
+    int[] heap;
+    int size;
+    int maxsize;
 
+    public static final int FRONT = 1;
+
+    public static void main(String[] args)
+    {
+        Minheap minHeap = new Minheap(15); 
+        minHeap.insert(10); 
+        minHeap.insert(7); 
+        minHeap.insert(6); 
+        minHeap.insert(12); 
+        minHeap.insert(17); 
+        minHeap.insert(5);
+        
+
+        System.out.println("min is "+minHeap.getmin());
+        System.out.println("the element removed is"+minHeap.remove());
+
+    }
+
+    public Minheap(int maxsize)
+    {
+        this.maxsize=maxsize;
+        this.size=0;
+        heap=new Int(this.maxsize+1);
+        heap[0]=Integer.MIN_VALUE;
+    } 
+
+    //parent node
+    private int parent(int pos)
+    {
+        return pos/2;
+    } 
+
+    //left
+    private int left(int pos)
+    {
+        return (2*pos);
+    }
+
+    //right
+    private int right(int pos)
+    {
+        return (2*pos)+1;
```

Comment:
also give explanation that why is this giving my right child 
so that i get to know that you are also aware of the logic behind it